test/mmapper/mmapper.c

```diff
@@ -0,0 +1,93 @@
+#define _GNU_SOURCE
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/mman.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <sched.h>
+#include <signal.h>
+#include <string.h>
+
+#ifndef CLONE_NEWPID
+#define CLONE_NEWPID 0x20000000
+#endif
+
+static int do_test(void *logf)
+{
+    int fd, i = 0;
+
+    // Detach from the parent terminal
+    setsid();
+
+    // Close standard file descriptors
+    close(STDIN_FILENO);
+    close(STDOUT_FILENO);
+    close(STDERR_FILENO);
+
+    // Open /dev/null as stdin
+    fd = open("/dev/null", O_RDONLY);
+    if (fd != STDIN_FILENO)
+    {
+        dup2(fd, STDIN_FILENO);
+        close(fd);
+    }
+
+    // Open the log file and redirect stdout and stderr to it
+    fd = open(logf, O_WRONLY | O_TRUNC | O_CREAT, 0600);
+    dup2(fd, STDOUT_FILENO);
+    dup2(fd, STDERR_FILENO);
+    if (fd != STDOUT_FILENO && fd != STDERR_FILENO)
+        close(fd);
+
+    while (1)
+    {
+        sleep(1);
+        printf("%d\n", i++);
+        fflush(stdout);
+    }
+
+    return 0;
+}
+
+int main(int argc, char *argv[])
+{
+    if (argc < 3)
+    {
```

Comment:
We tend to follow the kernel coding style:
https://www.kernel.org/doc/html/latest/process/coding-style.html#placing-braces-and-spaces

Reply:
Would it make sense to commit this [.clang-format](https://github.com/checkpoint-restore/criu/blob/criu-dev/.clang-format) configuration file from `criu` in this repository to help with C code formatting ?

Reply:
I'm not sure if this is necessary. With the exception of `piggie.c`, `loop.c`, and `mmapper.c`, we do not have C code the go-criu repository.

In the CRIU repository, we [fetch](https://github.com/checkpoint-restore/criu/blob/21992206b2e049f0e41efcfa46f9cdbba0622ca8/Makefile#L472) the `.clang-format` file from the kernel git repository and apply a few modifications:

https://github.com/checkpoint-restore/criu/blob/criu-dev/scripts/fetch-clang-format.sh

Reply:
I understand and agree that this is not that necessary here.